src/data/thunks.js

```diff
@@ -98,6 +99,27 @@ export function getProctoringSettings() {
   };
 }
 
+export function getExamAccessToken() {
+  return async (dispatch, getState) => {
+    const { exam } = getState().examState;
+    if (!exam.id) {
+      logError('Failed to get exam access token. No exam id.');
+      handleAPIError(
+        // todo: check what user error is here, change error?
```

Comment:
Question: should there be a more clear error?

Reply:
That's fair! Sounds good. Given this and the conversation about potentially just having an api function that returns the exam access token, I'm wondering if we should remove all of the logic from thunks and simply do the logic in the checkExamAccessToken function. Since we don't need to store the api error/exam access token (since we are just returning this), I think a thunk is probably not necessary. Since thunks typically just update the state and don't actually return anything? Or rather they technically return a function that updates state. 

What do you think?